terminal.integrated.killGracefully Setting has opposite effect on windows

Testing #249671

Went through the test steps with the setting disabled - no error

enabled the setting just to see if there was any difference

this time 

```
thread 'main' panicked at examples\hello-world\src\main.rs:17:10:
called `Result::unwrap()` on an `Err` value: Os { code: 10048, kind: AddrInUse, message: "Only one usage of each socket address (protocol/network address/port) is normally permitted." }
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
error: process didn't exit successfully: `target\debug\example-hello-world.exe` (exit code: 101)
Makefile:2: recipe for target 'do-cargo-run' failed
mingw32-make: *** [do-cargo-run] Error 101
```

note I had to use mingw32-make on windows ARM since make isn't available

--- a/src/vs/platform/terminal/node/terminalProcess.ts
+++ b/src/vs/platform/terminal/node/terminalProcess.ts
@@ -377,7 +377,8 @@ export class TerminalProcess extends Disposable implements ITerminalChildProcess
 			if (this._ptyProcess) {
 				await this._throttleKillSpawn();
 				this._logService.trace('node-pty.IPty#kill');
-				if (this.shellLaunchConfig.killGracefully) {
+				// For windows, we already attempt to gracefully kill the process
+				if (!isWindows && this.shellLaunchConfig.killGracefully) {
 					this._killGracefully(this._ptyProcess);
 				} else {
 					this._ptyProcess.kill();
@@ -391,19 +392,7 @@ export class TerminalProcess extends Disposable implements ITerminalChildProcess
 	}
 
 	private async _killGracefully(ptyProcess: IPty): Promise<void> {
-		if (!isWindows) {
-			ptyProcess.kill('SIGTERM');
-		} else if (isWindows && process.platform === 'win32') {
-			const windir = process.env['WINDIR'] || 'C:\\Windows';
-			const TASK_KILL = path.join(windir, 'System32', 'taskkill.exe');
-			try {
-				await exec(`${TASK_KILL} /T /PID ${ptyProcess.pid}`);
-			} catch (err) {
-				ptyProcess.kill();
-			}
-		} else {
-			ptyProcess.kill();
-		}
+		ptyProcess.kill('SIGTERM');
 	}
 
 	private async _throttleKillSpawn(): Promise<void> {

--- a/src/vs/platform/terminal/node/terminalProcess.ts
+++ b/src/vs/platform/terminal/node/terminalProcess.ts
@@ -377,8 +377,7 @@ export class TerminalProcess extends Disposable implements ITerminalChildProcess
 			if (this._ptyProcess) {
 				await this._throttleKillSpawn();
 				this._logService.trace('node-pty.IPty#kill');
-				// For windows, we already attempt to gracefully kill the process
-				if (!isWindows && this.shellLaunchConfig.killGracefully) {
+				if (this.shellLaunchConfig.killGracefully) {
 					this._killGracefully(this._ptyProcess);
 				} else {
 					this._ptyProcess.kill();

--- a/src/vs/workbench/contrib/terminal/browser/terminalInstance.ts
+++ b/src/vs/workbench/contrib/terminal/browser/terminalInstance.ts
@@ -567,7 +567,8 @@ export class TerminalInstance extends Disposable implements ITerminalInstance {
 				this._setAriaLabel(this.xterm?.raw, this._instanceId, this.title);
 			}
 			if (e.affectsConfiguration(TerminalSettingId.KillGracefully)) {
-				this.shellLaunchConfig.killGracefully = this._configurationService.getValue(TerminalSettingId.KillGracefully);
+				// For windows, we already attempt to gracefully kill the process
+				this.shellLaunchConfig.killGracefully = !isWindows && this._configurationService.getValue(TerminalSettingId.KillGracefully);
 			}
 			if (e.affectsConfiguration('terminal.integrated')) {
 				this.updateConfig();
@@ -1509,7 +1510,8 @@ export class TerminalInstance extends Disposable implements ITerminalInstance {
 				message: nls.localize('workspaceNotTrustedCreateTerminalCwd', "Cannot launch a terminal process in an untrusted workspace with cwd {0} and userHome {1}", this._cwd, this._userHome)
 			});
 		}
-		this.shellLaunchConfig.killGracefully = this._configurationService.getValue(TerminalSettingId.KillGracefully);
+		// For windows, we already attempt to gracefully kill the process
+		this.shellLaunchConfig.killGracefully = !isWindows && this._configurationService.getValue(TerminalSettingId.KillGracefully);
 		// Re-evaluate dimensions if the container has been set since the xterm instance was created
 		if (this._container && this._cols === 0 && this._rows === 0) {
 			this._initDimensions();

--- a/src/vs/workbench/contrib/terminal/common/terminalConfiguration.ts
+++ b/src/vs/workbench/contrib/terminal/common/terminalConfiguration.ts
@@ -645,7 +645,7 @@ const terminalConfiguration: IConfigurationNode = {
 			default: false,
 			markdownDescription: localize(
 				'terminal.integrated.killGracefully',
-				'Controls how terminal processes are terminated. When enabled, attempts a graceful shutdown of the process tree. On POSIX, this will send \`SIGTERM\`. On Windows, it uses \`taskkill.exe\` without the \`/F\` (force) flag. Note that this may allow misbehaving processes to remain running. When disabled (default), processes are terminated with: \`SIGHUP\`.'
+				'Controls how terminal processes are terminated on macOS and Linux. When enabled, attempts a graceful shutdown of the process tree. This will send \`SIGTERM\`. When disabled (default), processes are terminated with: \`SIGHUP\`.'
 			),
 		},
 		...terminalContribConfiguration,

